File: python code/basic functions/readwritedisplay.py
import cv2

image = cv2.imread("../../images/lena.jpg")

# check for invalid input
if image is None:
    print("Could not open or find the image")

# convert color image to gray
grayImage = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)

# save result
cv2.imwrite("imageGray.jpg", grayImage)

# create a window for display
cv2.namedWindow("Image", cv2.WINDOW_AUTOSIZE)
cv2.namedWindow("Gray Image", cv2.WINDOW_NORMAL)

# display image
cv2.imshow("Image", image)
cv2.imshow("Gray Image", grayImage)

# wait for a keystroke in window
cv2.waitKey(0)

File: c++ code/week1/examples/basic functions/basic functions/readwritedisplay.cpp
#include <opencv2/opencv.hpp>

int main1()
{
	// read image
	cv::Mat image = cv::imread("../../../../../images/lena.jpg", cv::IMREAD_ANYCOLOR);

	// check for invalid input
	if (image.empty()) {
		std::cout << "Coutld not open or find image\n";
		return EXIT_FAILURE;
	}

	// convert color image to gray
	cv::Mat grayImage;
	cv::cvtColor(image, grayImage, cv::COLOR_BGR2GRAY);

	// create a window for display
	cv::namedWindow("Image", CV_WINDOW_AUTOSIZE);
	cv::namedWindow("Gray Image", CV_WINDOW_NORMAL);

	// display image
	cv::imshow("Image", image);
	cv::imshow("Gray Image", grayImage);

	// wait for a keystroke in the window
	cv::waitKey(0);

	return EXIT_SUCCESS;
}